docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1135 @@
+
+**Design considerations**
+
+Parsing of the user’s input command:
+
+- Alternative 1 (current choice): User’s command is split into size 2 array first containing command type and command arguments. Then arguments are split into command tag and information pairs.  
+
+|     |     |
+|-----|-----|
+|**Pros** | Command tags do not have to follow a fixed order.|
+|**Cons** | It takes multiple steps in parsing the command.|
+
+- Alternative 2: User’s command is divided by space
+                 
+|     |     |
+|-----|-----|
+|**Pros** | The parsing can be easily done by calling Java built-in function .split(). Supports multiple tags or no tags.|
+|**Cons** | Values for each variable cannot contain spaces which makes the application restrictive, especially for user's name.|
+
+[&#8593; Return to Top](#developer-guide)
+
+#### 4.2.2. <a id="viewing-a-profile">Viewing a Profile</a>
+This feature allows user to view added profile with calculated BMI based on height and weight. The failure to do so will trigger an exception where the user will be notified of the reason, e.g. redundant parameters. The action will be aborted, and the program will advise the user to type "help" for command syntax reference. 
+
+If the data loading is successful, a message on the added profile will be displayed to the user. 
+
+**Implementation**
+
+When the user attempts to view an added profile, the ProfileSession, Ui, ProfileParser, Command, CommandLib, ProfileStorage, Profile, DietManager and CommandResult classes will be accessed. The following sequence of steps will then occur:
+
+1. User executes `view`
+    1. `ProfileSession` calls `Ui.getUserCommand()` to receive user input.
+    1. ProfileSession` calls `ProfileParser.parseCommand()` to parse user input into a string array.
+1. Creating `ProfileView` object.
+    1. Based on the parsed input, `ProfileSession` calls `CommandLib` to return the correct Command Object `ProfileView`.
+1. Executing command.
+    1. `ProfileSession` calls `ProfileView.execute()` with the rest of parsed input.
+    1. `ProfileView` calls `ProfileStorage.loadData()` to load existing profile in the system. If there is no existing profile, `ProfileView` returns a failure result to `ProfileSession`. Otherwise, the process continues with step `iii`.
+    1. `ProfileView` calls `DietManager.getTodayTotalCalories()` to get user's calories intake today.
+    1. Based on user's calories intake today and string representation of `Profile`, `ProfileView` returns a result to `ProfileSession`.    
+1. Prompting result to user.
+    1. `ProfileSession` calls `CommandResult.getCommandResult()` to get the `CommandResult` object.
+    1. `ProfileSession` calls `Ui.showToUser()` to show result to the user.
+
+All descriptions, warnings and responses will be handled by `Ui` to ensure consistence across the app.
+
+The sequence diagram below summarizes how viewing an added profile works:
+
+![Load Data Sequence Diagram](pictures/khoa/ViewProfile.png)
```

Comment:
![CS2113(2)](https://user-images.githubusercontent.com/65339123/97544581-cb6c5d00-1a04-11eb-9268-0a0657c1d28d.PNG)

I believe the 'X' should be extended instead of just directly placed at the end of the activation bar